compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -313,21 +313,35 @@ impl<'a> FunctionContext<'a> {
             }
         };
 
-        if let Some(max_bit_size) = operator_result_max_bit_size_to_truncate(
+        // Check for integer overflow
+        if matches!(
             operator,
-            lhs,
-            rhs,
-            &self.builder.current_function.dfg,
+            BinaryOpKind::Add
+                | BinaryOpKind::Subtract
+                | BinaryOpKind::Multiply
+                | BinaryOpKind::ShiftLeft
```

Comment:
Since `insert_binary` is getting large, can we break out this `if` block into another function like `check_for_overflow`?